src/zig.rs

```diff
@@ -166,6 +168,128 @@ impl zed::Extension for ZigExtension {
+                    .collect();
+                args.push("--test-no-exec".into());
+                match zed::current_platform().0 {
+                    zed::Os::Windows => args.push(format!("-femit-bin='{test_exe_path}.exe'")),
+                    _ => args.push(format!("-femit-bin={test_exe_path}")),
+                }
+
+                zed::BuildTaskTemplate {
+                    label: "zig test --test-no-exec".into(),
+                    command: "zig".into(),
+                    args,
+                    env,
+                    cwd,
+                }
+            }
+            _ => return None,
+        };
+
+        let config = serde_json::Value::Null;
+        let Ok(config) = serde_json::to_string(&config) else {
+            return None;
+        };
+
+        Some(zed::DebugScenario {
+            adapter: debug_adapter_name,
+            label: resolved_label.clone(),
+            config,
+            tcp_connection: None,
+            build: Some(zed::BuildTaskDefinition::Template(
+                zed::BuildTaskDefinitionTemplatePayload {
+                    template,
+                    locator_name: Some(locator_name.into()),
+                },
+            )),
+        })
+    }
+
+    fn run_dap_locator(
+        &mut self,
+        _locator_name: String,
+        build_task: zed::TaskTemplate,
+    ) -> Result<zed::DebugRequest, String> {
+        let mut args_it = build_task.args.iter();
+        match args_it.next() {
+            Some(arg) if arg == "build" => {
+                // We only handle the default case where the binary name matches the project name.
+                // This is valid for projects created with `zig init`.
+                // In other cases, the user should provide a custom debug configuration.
+                let exec = get_project_name(&build_task).ok_or("Failed to get project name")?;
+
+                let request = zed::LaunchRequest {
+                    program: format!("zig-out/bin/{exec}"),
+                    cwd: build_task.cwd,
+                    args: vec![],
+                    envs: build_task.env.into_iter().collect(),
+                };
+
+                Ok(zed::DebugRequest::Launch(request))
+            }
+            Some(arg) if arg == "test" => {
+                let program = get_test_exe_path().unwrap();
+                let request = zed::LaunchRequest {
+                    program,
+                    cwd: build_task.cwd,
+                    args: vec![],
+                    envs: build_task.env.into_iter().collect(),
+                };
+                Ok(zed::DebugRequest::Launch(request))
+            }
+            _ => Err("Unsupported build task".into()),
+        }
+    }
+}
+
+fn get_project_name(task: &zed::TaskTemplate) -> Option<String> {
+    task.cwd
+        .as_ref()
+        .and_then(|cwd| Some(Path::new(&cwd).file_name()?.to_string_lossy().into_owned()))
+}
+
+fn get_test_exe_path() -> Option<String> {
```

Comment:
This will trip over itself the moment somebody runs two tests (possibly in unrelated projects) concurrently. :/ Could we include some kind of disambiguator here?

Reply:
@osiewicz addressed in 6ef6332